extensions/import/src/wizard/pages/fileConfigPage.ts

```diff
@@ -0,0 +1,393 @@
+			actions: [this.fileButton]
+		};
+	}
+
+	private async createTableNameBox(): Promise<sqlops.FormComponent> {
+		this.tableNameTextBox = this.view.modelBuilder.inputBox().withValidation((name) => {
+			let tableName = name.value;
+
+			if (!tableName || tableName.length === 0) {
+				return false;
+			}
+
+			if (this.tableNames.indexOf(tableName) !== -1) {
+				return false;
+			}
+
+			return true;
+		}).component();
+
+		this.tableNameTextBox.onTextChanged((tableName) => {
+			this.model.table = tableName;
+		});
+
+		return {
+			component: this.tableNameTextBox,
+			title: localize('flatFileImport.tableTextboxTitle', 'New table name'),
+		};
+	}
+
+
+	private async createSchemaDropdown(): Promise<sqlops.FormComponent> {
+		this.schemaDropdown = this.view.modelBuilder.dropDown().component();
+		this.schemaLoader = this.view.modelBuilder.loadingComponent().withItem(this.schemaDropdown).component();
+
+		this.schemaDropdown.onValueChanged(() => {
+			this.model.schema = (<sqlops.CategoryValue>this.schemaDropdown.value).name;
+		});
+
+
+		return {
+			component: this.schemaLoader,
+			title: localize('flatFileImport.schemaTextboxTitle', 'Table schema'),
+		};
+
+	}
+
+	private async populateSchemaDropdown(): Promise<Boolean> {
+		this.schemaLoader.loading = true;
+		let connectionUri = await sqlops.connection.getUriForConnection(this.model.server.connectionId);
+		let queryProvider = sqlops.dataprotocol.getProvider<sqlops.QueryProvider>(this.model.server.providerName, sqlops.DataProviderType.QueryProvider);
+
+		let query = `SELECT name FROM sys.schemas`;
+
+		let results = await queryProvider.runQueryAndReturn(connectionUri, query);
+
+		let idx = -1;
+		let count = -1;
+
+		let values = results.rows.map(row => {
+			let schemaName = row[0].displayValue;
+			count++;
+			if (this.model.schema && schemaName === this.model.schema) {
+				idx = count;
+			}
+			let val = row[0].displayValue;
+
+			return {
+				name: val,
+				displayName: val
+			};
+		});
+
+		if (idx > 0) {
+			let tmp = values[0];
+			values[0] = values[idx];
+			values[idx] = tmp;
+		}
+
+		this.model.schema = values[0].name;
+
+		this.schemaDropdown.updateProperties({
+			values: values
+		});
+
+		this.schemaLoader.loading = false;
+		return true;
+	}
+
+	private async populateTableNames(): Promise<boolean> {
+		this.tableNames = [];
+		let databaseName = (<sqlops.CategoryValue>this.databaseDropdown.value).name;
+
+		if (!databaseName || databaseName.length === 0) {
+			this.tableNames = [];
+			return false;
+		}
+
+		let connectionUri = await sqlops.connection.getUriForConnection(this.model.server.connectionId);
+		let queryProvider = sqlops.dataprotocol.getProvider<sqlops.QueryProvider>(this.model.server.providerName, sqlops.DataProviderType.QueryProvider);
+		let results: sqlops.SimpleExecuteResult;
+
+		try {
+			let query = sqlstring.format('USE ?; SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = \'BASE TABLE\'', [databaseName]);
```

Comment:
This query appears to be broken for me now. I don't think `sqlstring` is going to be able to do the right thing for us here

Reply:
Yep, I see the same issue on my side. What do you suggest?